system/jlib/jutil.cpp

```diff
@@ -1665,15 +1687,21 @@ unsigned msTick()
 #warning "clock_gettime(CLOCK_MONOTONIC) not supported"
 unsigned msTick()
 {
-  struct timeval tm;
-  gettimeofday(&tm,NULL);
-  return tm.tv_sec*1000+(tm.tv_usec/1000);
+    struct timeval tm;
+    gettimeofday(&tm,NULL);
+    return tm.tv_sec*1000+(tm.tv_usec/1000);
 }
 unsigned usTick()
 {
-  struct timeval tm;
-  gettimeofday(&tm,NULL);
-  return tm.tv_sec*1000000+tm.tv_usec;
+    struct timeval tm;
+    gettimeofday(&tm,NULL);
+    return tm.tv_sec*1000000+tm.tv_usec;
+}
+unsigned __int64 nsTick()
+{
+    struct timeval tm;
+    gettimeofday(&tm,NULL);
+    return tm.tv_sec*1000000000+tm.tv_usec*1000;
```

Comment:
same question here

Reply:
Got catch - I think this is a real problem.

Reply:
Your comments suggest you had changed this in latest push, but it appears unchanged.

Reply:
Good catches.  Fixed.